pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -418,36 +473,69 @@ class Win32DeploymentReader {
+ * @param profile The profile to be validated, modified in place
+ * @param errors An array of error messages, built up in place
+ * @param schema The structure (usually defaultSettings) used as a template
+ * @param parentPathParts The parent path for the current schema key.
+ * @returns The original profile, less any invalid fields
+ */
+function validateDeploymentProfileWithErrors(profile: any, errors: string[], schema: any, parentPathParts: string[]) {
   if (typeof profile !== 'object') {
     return profile;
   }
+  const fullPath = (key: string) => {
+    return parentPathParts.length === 0 ? key : `${ parentPathParts.join('.') }.${ key }`;
+  };
+
   for (const key in profile) {
     if (!(key in schema)) {
-      console.log(`Deployment Profile ignoring '${ parentPathParts.join('.') }.${ key }': not in schema.`);
+      console.log(`Deployment Profile ignoring '${ fullPath(key) }': not in schema.`);
       delete profile[key];
       continue;
     }
-    if (typeof profile[key] !== 'object') {
-      if (typeof profile[key] !== typeof schema[key]) {
-        console.log(`Deployment Profile ignoring '${ parentPathParts.join('.') }.${ key }': expecting value of type ${ typeof schema[key] }, got ${ typeof profile[key] }.`);
-        delete profile[key];
+    const schemaVal = schema[key];
+    const profileVal = profile[key];
+
+    if (Array.isArray(profileVal) || Array.isArray(schemaVal)) {
+      if (Array.isArray(profileVal) !== Array.isArray(schemaVal)) {
+        if (Array.isArray(schemaVal)) {
+          errors.push(`Error for field '${ fullPath(key) }': expecting an array, got '${ JSON.stringify(profileVal) }'`);
+        } else {
+          errors.push(`Error for field '${ fullPath(key) }': expecting value of type ${ typeof schemaVal }, got an array ${ JSON.stringify(profileVal) }`);
+        }
       }
-    } else if (Array.isArray(profile[key])) {
-      if (!Array.isArray(schema[key])) {
-        console.log(`Deployment Profile ignoring '${ parentPathParts.join('.') }.${ key }': got an array, expecting type ${ typeof schema[key] }.`);
-        delete profile[key];
+    } else if (typeof profileVal !== 'object') {
+      if (typeof profileVal !== typeof schemaVal) {
+        errors.push(`Error for field '${ fullPath(key) }': expecting value of type ${ typeof schemaVal }, got '${ JSON.stringify(profileVal) }'`);
       }
     } else if (haveUserDefinedObject(parentPathParts.concat(key))) {
       // Keep this part of the profile
+    } else if (typeof profileVal !== typeof schemaVal) {
+      errors.push(`Error for field '${ fullPath(key) }': expecting value of type ${ typeof schemaVal }, got '${ JSON.stringify(profileVal) }'`);
     } else {
       // Finally recurse and compare the schema sub-object with the specified sub-object
-      validateDeploymentProfile(profile[key], schema[key], parentPathParts.concat(key));
+      const newParentParts = parentPathParts.length === 0 ? [key] : parentPathParts.concat(key);
```

Comment:
```suggestion
      const newParentParts = [...parentPathParts, key];
```
But at that point we don't even need the temporary anymore.